### PR TITLE
Make adjustments to support PHP8.1's stricter type system

### PR DESCRIPTION
**Pull request by nezaniel**
This adds additional type hints where required in PHP 8.1 while maintaining support for 7.4.
These are only the adjustments required fir the ESCR test suite, more are to follow

### CODE REVIEW

**Review by albe — APPROVED** [review]
Looks good by reading, but I agree having 8.1 builds run would be good

**Comment by nezaniel** [comment]
8.1 builds are configured and run through smoothly, which leaves me a bit sceptical since that seems improbable. Is there more to be configured or is everything fine already?

**Comment by albe** [comment]
> 8.1 builds are configured and run through smoothly, which leaves me a bit sceptical since that seems improbable. Is there more to be configured or is everything fine already?

> ==> Setup PHP
✓ PHP Found PHP 8.1.2

Looks good to me. Are there any specific reasons why you think running Flow on 8.1 should not work with this PR applied @nezaniel ? Are there any breaking changes in 8.1 that we should be subject to?
I could very well imagine Flow being okay and Neos having a few more edge cases from a gut feeling

**Comment by mficzel** [comment]
i somehow miss the psalm test in the matrix now. Should offcourse not run on 7.3 but on 8.0 i think.

**Comment by nezaniel** [comment]
@albe Well, I had to un-proxy literally every ESCR exception class for it to not throw a __wakeup() signature notice, so I expected this to happen all over the Flow code base... but maybe the ESCR test setup is a little more strict than the regular tests and PHP runtimes

As far as I could see, the psalm tests were configured for 7.3 only, so I disabled them

File: Neos.Flow/Tests/Behavior/Features/Bootstrap/SecurityOperationsTrait.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ public function iHaveTheFollowingPolicies($string)
      */
     public static function cleanUpSecurity()
     {
-        if (file_exists(self::$testingPolicyPathAndFilename)) {
+        if (self::$testingPolicyPathAndFilename && file_exists(self::$testingPolicyPathAndFilename)) {
```
Comment (bwaidelich) [inline Neos.Flow/Tests/Behavior/Features/Bootstrap/SecurityOperationsTrait.php:107]: nitpick: This is a "falsy" value but we prefer strict comparison according to the [CGL](https://neos.readthedocs.io/en/stable/References/CodingGuideLines/PHP.html#php-in-general)
```suggestion
        if (self::$testingPolicyPathAndFilename !== null && file_exists(self::$testingPolicyPathAndFilename)) {
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -16,16 +16,16 @@ jobs:
...
-        include:
-          - php-versions: '7.3'
-            static-analysis: 'psalm'
-            experimental: true
-            dependencies: 'highest'
+        #include:
+          #- php-versions: '7.3'
+          #  static-analysis: 'psalm'
+          #  experimental: true
+          #  dependencies: 'highest'
```
Comment (bwaidelich) [inline .github/workflows/build.yml:28]: I'd say: remove those lines.. If we want them back, we can always look them up in the git history

**Review by bwaidelich — APPROVED** [review]
Looking good in general, I just added to nitpicky comments

File: .github/workflows/build.yml (hunk 2 above)
Comment (nezaniel) [reply .github/workflows/build.yml:28]: Since iirc the psalm checks failed before anyway, I'd like to leave them out for now, as I'd like this merged independently of us getting psalm back to life. Also, I never worked with psalm before, so I won't be able to help here

Comment (albe) [reply .github/workflows/build.yml:28]: Okay, we'll deal with that in #2704
